aspnetcore/client-side/import-export-interop.md

```diff
@@ -0,0 +1,394 @@
+    <WasmExtraFilesToDeploy Include="*.css" />
+  </ItemGroup>
+
+</Project>
+```
+
+## .NET JavaScript interop on :::no-loc text="WASM":::
+
+APIs in the following example are imported from `dotnet.js`. These APIs enable you to set up named modules that can be imported into your C# code and call into methods exposed by your .NET code, including `Program.Main`.
+
+> [!IMPORTANT]
+> "Import" and "export" throughout this article are defined from the perspective of .NET:
+>
+> * An app imports JS methods so that they can be called from .NET.
+> * The app exports .NET methods so that they can be called from JS.
+
+Function calls in the following example:
+
+* `dotnet.create()` sets up the .NET WebAssembly runtime.
+* `setModuleImports` creates the `window.location.href` function, which returns the current page address (URL). The import designates a module name, which must match the name used with the `JSImportAttribute` (explained later in this article). The following example uses the module name `main.js`, which is merely a convention for the example based on the filename `main.js`. The `window.location.href` function is imported into C# and called by the C# method `GetHRef`. The `GetHRef` method is shown later in this section.
+* `exports.MyClass.Greeting()` calls into .NET (`MyClass.Greeting`) from JS. The `Greeting` C# method returns a string that includes the result of calling the `window.location.href` function. The `Greeting` method is shown later in this section.
+* `runMainAndExit` runs `Program.Main`.
+
+<!--
+
+    NOTE: THE FOLLOWING CODE IS CHANGING FOR RELEASE:
+
+    https://github.com/dotnet/runtime/blob/release/7.0/src/mono/wasm/templates/templates/browser/main.js
+
+    For example, the runMainAndExit() is changing to dotnet.run().
+
+    This is tracked on the 7.0 doc tracking issue at https://github.com/dotnet/AspNetCore.Docs/issues/26364.
+    
+-->
+
+`main.js` of an app created from the `wasmbrowser` project template:
+
+```javascript
+import { dotnet } from './dotnet.js'
+
+const is_browser = typeof window != "undefined";
+if (!is_browser) throw new Error(`Expected to be running in a browser`);
+
+const { setModuleImports, getAssemblyExports, getConfig, runMainAndExit } = 
+  await dotnet.create();
+
+setModuleImports("main.js", {
+  window: {
+    location: {
+      href: () => globalThis.window.location.href
+    }
+  }
+});
+
+const config = getConfig();
+const exports = await getAssemblyExports(config.mainAssemblyName);
+const text = exports.MyClass.Greeting();
+console.log(text);
+
+document.getElementById("out").innerHTML = `${text}`;
+await runMainAndExit(config.mainAssemblyName, ["dotnet", "is", "great!"]);
+```
+
+> [!NOTE]
+> You can inspect the .NET host builder APIs at the [`dotnet/runtime` GitHub repository](https://github.com/dotnet/runtime/blob/main/src/mono/wasm/runtime/). The runtime configuration is in the [`dotnet.d.ts` file](https://github.com/dotnet/runtime/blob/main/src/mono/wasm/runtime/dotnet.d.ts).
```

Comment:
How about we just link to dotnet.d.ts in the Additional resources section at the end of this article?
```suggestion
```